### ISSUE TITLE
Providing new information to LLM03: Training Data Poisoning

### ISSUE THREAD

**Issue by ManuelSLemos** [OWASP/www-project-top-10-for-large-language-model-applications#205]
Hi, this is my first issue in an owasp repository, so I hope I'm not wrong with the style guide. In my company we are studying different ways of pentesting AI, specifically LLMs, and I would like to share that knowledge. The additions are:

## Common Examples of Vulnerability

4. Using data provided by users without any quality control for training or fine-tuning purposes
 a.  A malicious actor individually or collectively (e.g. botnet) adds poisonous information in the form of prompts. 
 b. A malicious actor individually or collectively (e.g. botnet) uses the feedback systems (voting) against the model to add poisonous information.

## How to Prevent

7. e. Elaborate control questions related to our business to ensure that the model has not been poisoned and integrate it into the "MLSecOps" cycle.

8. Use DVC (Data Version Control) to keep track of which part of the dataset has been manipulated, deleted or new data added.

9. Use Vector Database to add user-supplied information, to protect from poisoning other users and even fix in production without having to re-train a new mode

10. Verify the training dataset used when choosing a public model to avoid using a previously poisoned model.

That's all, I look forward to your feedback and if you like the contribution, I will make a PR adding the content. Thank you very much.

**Comment by ManuelSLemos** [comment]
For some reason, it won't let me add a label, sorry. @GangGreenTemperTatum

**Comment by ManuelSLemos** [comment]
Hi @GangGreenTemperTatum!

Thank you very much for the feedback, I'll get back to you:

> Do you have any example resources that support this at all?

There are no articles that talk about these techniques directly, they are methods that I have read indirectly and in my company we have created from scratch.

* The control questions are inspired by the [tests ](https://github.com/EleutherAI/lm-evaluation-harness?tab=readme-ov-file)that use the LLM [leadboards](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard). I have developed new tests for each LLM we have in the company.

* For [DVCs](https://dvc.org/doc/user-guide/analytics), its own documentation explains how to analyze the different versions of datasets. The best feature it has is the ability to revert to a version of the dataset that we know is not poisoned. 

* And finally, vector databases. As in the DVC, in the documentation of different types of databases they talk about analyzing the embedded data and in addition, I use [pgvector ](https://github.com/pgvector/pgvector) to be able to separate the prompts by users to elaborate more personalized answers.

As the focus is on closing version 1_1, feel free to talk to me on Slack if I can help you in any way. :)

**Comment by ManuelSLemos** [comment]
Hello, I will explain these points better.

In those points I am trying to say that if you use user prompts for future training, a malicious actor can take advantage and add poisoned data. Just like if you don't protect the feedback system that evaluates the response, they can use it to confuse the model in the long run.  

By mentioning the leadboard, I meant that just as they have tests to measure benchmarking, I have created different tests for each LLM to measure "poisoning", "hallucinations" or simply the quality of the answers. It is not that I use that leadboard, but I was inspired by them to create my own tests.

I hope this explains it better and thank you for your patience.  :)

**Comment by ManuelSLemos** [comment]
Okay, that's right. Then I delegate to your discretion that it can be leveraged for a future version. Likewise, I will search/write about these techniques to add them as support material.  

Thank you very much!
